Add waypoint config panel to mission planning

ArturoManzoli opened the pull request:
To be merged after #1469 

Adds a starting version of the Waypoint config panel, as part of the "Extend survey to generic MAVLink commands" implementation


https://github.com/user-attachments/assets/cd2a9e8f-809a-4fd1-b024-82f5e26c35d8

rafaellehmkuhl commented:
Hey Arturo, could you rebase this branch over master?

ArturoManzoli commented:
> Hey Arturo, could you rebase this branch over master?
> 
> As it's based on #1469, you should actually rebase that one over master and them rebase this one over #1469.

Yep, I did push the wrong stuff. 
Now it has only the diff to be added after merging #1469

rafaellehmkuhl commented:
> > Hey Arturo, could you rebase this branch over master?
> > As it's based on #1469, you should actually rebase that one over master and them rebase this one over #1469.
> 
> Yep, I did push the wrong stuff. Now it has only the diff to be added after merging #1469



> > Hey Arturo, could you rebase this branch over master?
> > As it's based on #1469, you should actually rebase that one over master and them rebase this one over #1469.
> 
> Yep, I did push the wrong stuff. Now it has only the diff to be added after merging #1469

Much better! This will be a lot easier to review!

One thing: this seems to be rebased over master, not over the branch of #1469 (I can see that because the changes of #1469 are not here), but we can keep it this way as it will be easier to review, and rebase over master once we merge #1469.

rafaellehmkuhl commented on src/components/mission-planning/ContextMenu.vue:
```diff
@@ -82,23 +82,91 @@
-        <span class="text-white text-sm ml-4">{{ surveyCreationButtonText }}</span>
-      </v-list-item>
+      <div class="flex justify-between items-center pt-1 pb-2 px-2">
+        <p class="text-[14px]">Waypoint {{ missionStore.getWaypointNumber(selectedWaypoint?.id as string) }}</p>
```
The creation of the `missionStore.getWaypointNumber(...)` should be done in this commit.

ArturoManzoli replied:
done

rafaellehmkuhl commented on src/views/MissionPlanningView.vue:
```diff
@@ -1515,8 +1708,20 @@ onMounted(async () => {
+  planningMap.value.on('contextmenu', (e: LeafletMouseEvent) => {
+    if (isCreatingSurvey.value) return
+    selectedWaypoint.value = undefined
+    if (selectedSurveyId.value === '') {
+      contextMenuType.value = 'map'
+      showContextMenu(e)
+    }
+    if (selectedSurveyId.value !== '') {
+      showContextMenu(e)
+    }
```
This expression can be simplified since `showContextMenu(e)` will always be called in the end.

rafaellehmkuhl commented on src/components/mission-planning/WaypointConfigPanel.vue:
```diff
@@ -0,0 +1,169 @@
+  emit('removeWaypoint', waypoint)
+}
+
+const availableFrames = [
```
This should be typed using the `AltitudeReferenceType` enum so we never leave one left.

rafaellehmkuhl commented on src/components/mission-planning/WaypointConfigPanel.vue:
```diff
@@ -0,0 +1,169 @@
+      :is-expanded="true"
+    >
+      <template #title
+        ><p class="ml-4 text-center text-[13px] font-normal">
+          {{
+            selectedWaypoint.id === 'home'
+              ? 'Home'
+              : `Waypoint  ${missionStore.getWaypointNumber(selectedWaypoint?.id as string)}
+          parameters`
+          }}
+        </p></template
+      >
```
Those indentations can be better formatted.

rafaellehmkuhl commented on src/views/MissionPlanningView.vue:
```diff
@@ -1042,11 +1138,11 @@ const updateSurveyEdgeAddMarkers = (): void => {
-              <path d="M10 5V15M5 10H15" stroke="#3B82F6" stroke-width="2"/>
-            </svg>
-          `,
+              <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg" style="display: block;">
+                <circle cx="10" cy="10" r="9" fill="white" stroke="#3B82F6" stroke-width="2"/>
+                <path d="M10 5V15M5 10H15" stroke="#3B82F6" stroke-width="2"/>
+              </svg>
+            `,
```
No need to reindent.

rafaellehmkuhl commented on src/views/MissionPlanningView.vue:
```diff
@@ -1259,21 +1377,21 @@ const createSurveyVertexMarker = (
   return L.marker(latlng, {
     icon: L.divIcon({
       html: `
-        <div class="survey-vertex-icon">
-          <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
-            <circle cx="12" cy="12" r="5" fill="#3B82F6" stroke="white" stroke-width="2"/>
-          </svg>
-          <div class="delete-popup" style="display: none;">
-            <button class="delete-button">
-              <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
-                <path d="M2 4h12M4 4v10a2 2 0 002 2h4a2 2 0 002-2V4M6 4V2h4v2" 
-                      stroke="white" stroke-width="1.5" 
-                      stroke-linecap="round" stroke-linejoin="round"/>
-                </svg>
-              </button>
+          <div class="survey-vertex-icon">
+            <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+              <circle cx="12" cy="12" r="5" fill="#3B82F6" stroke="white" stroke-width="2"/>
+            </svg>
+            <div class="delete-popup" style="display: none;">
+              <button class="delete-button">
+                <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+                  <path d="M2 4h12M4 4v10a2 2 0 002 2h4a2 2 0 002-2V4M6 4V2h4v2" 
+                        stroke="white" stroke-width="1.5" 
+                        stroke-linecap="round" stroke-linejoin="round"/>
+                  </svg>
+                </button>
+            </div>
           </div>
-        </div>
-      `,
+        `,
```
No need to reindent.

It looks like a lot of leftovers from the mission planning PR were readded here.

rafaellehmkuhl reviewed (changes requested):
Initial code review. Didn't test the changes extensively yet.

ArturoManzoli commented on src/components/mission-planning/ContextMenu.vue:
```diff
@@ -120,7 +188,7 @@ import { computed, defineEmits, defineProps } from 'vue'
 import { useAppInterfaceStore } from '@/stores/appInterface'
 import { useMissionStore } from '@/stores/mission'
-import { Survey } from '@/types/mission'
+import { ContextMenuTypes, Survey, Waypoint } from '@/types/mission'
```
done

rafaellehmkuhl reviewed (changes requested):
https://github.com/user-attachments/assets/547a1e76-b12c-4210-bc64-df86b99b4aeb

It appears like there's a bug in this side menu.

rafaellehmkuhl commented:
It seems like this patch is also breaking creation of simple path waypoints:


https://github.com/user-attachments/assets/ae621513-de0c-43c9-9122-893f2707b719

ArturoManzoli commented:
On last patch:
* Fixed visual bug on side menu;
* Fixed WP adding bug;
* Limited coordinates to 7 significant digits, so its under RTK gps resolution;
* Modified the mouse cursor to a '+' when inside the add simple path mode;

rafaellehmkuhl reviewed (changes requested):
I'm clicking a waypoint to configure it but it doesn't seem to allow me anymore. It did in the previous version.

<img width="283" alt="image" src="https://github.com/user-attachments/assets/3c9c3c43-2896-4c59-8ea2-aa50d1cf3125" />

ArturoManzoli commented:
> I'm clicking a waypoint to configure it but it doesn't seem to allow me anymore. It did in the previous version.
> 
Fixed!
It was related with the 7 decimal places display limit on coordinates.

ArturoManzoli commented:
@rafaellehmkuhl  
On last patch:

- Fixed WP selection not working outside survey and path creation modes;
- Updated button text to match 'simple path creation mode' closing;
